### DISCUSSION TITLE
Increase/Disable the timeout of the coffee editor

### DISCUSSION THREAD

**Discussion by Deel96** [eclipse-emfcloud/emfcloud#170] [Coffee Editor] [answered]
Hi,
the coffee-editor example disconnects after 30 minutes.
This also happens if you run it yourself.
I couldn't find the place where the timeout duration is specified / where it can be turned off.

In the theia-cloud project I saw something about "timeout limit" and "timeout strategy" but couldn't find any occurrence in the coffee-editor project except the timeout dialog.

Is it possible to change the timeout/ disable it?

Best regards.

**Comment by lucas-koehler** [comment, marked as answer]
Hi @Deel96,
the coffee editor should not timeout itself when you run it on your local machine because the timeout is enforced by the Kubernetes cluster. Are you sure it really timed out exactly after 30 minutes or could the connection to the backend be interrupted by some other incident (e.g. restarting it)?

The timeout dialog opens anytime the connection status of the coffee editor changes to offline. This is implemented in `web/coffee-server/src/browser/coffee-server-frontend-module.ts`.

To be clear, you run just the coffee editor, right? Or do you host it locally with theia-cloud or something similar?

**Reply by Deel96** [reply]
Hi @lucas-koehler,
I deployed it on che and theia-cloud.
Okay, then i know that something is crashing on the serverside.
Thank you for the quick reply !
